### ISSUE TITLE
Question: creating new rule, how to use PropertyAssignment 

### ISSUE THREAD

**Issue by retorquere** [palantir/tslint#3324]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import * as Lint from 'tslint'
import * as ts from 'typescript'

// The walker takes care of all the work.
class NoModuleExportsWalker extends Lint.RuleWalker {
  public visitPropertyAssignment(node: ts.PropertyAssignment) {
    // create a failure at the current position
    // this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING))

    console.log(node) // tslint:disable-line:no-console

    // call the base version of this visitor to actually parse this node
    super.visitPropertyAssignment(node)
  }
}

export class Rule extends Lint.Rules.AbstractRule {
  public static FAILURE_STRING = 'module.exports forbidden'

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    return this.applyWithWalker(new NoModuleExportsWalker(sourceFile, this.getOptions()))
  }
}
```

with `tslint.json` configuration:

```json
{ "rules": { "no-module-exports": true, } }
```

This is not a bug but an information request -- I'm trying to create a rule that forbids assigning to `module.exports` (as I want to enforce using `export =` instead), but I don't know how I can get whether it's `module.exports` being assigned to from inspecting the node.

**Comment by ajafff** [comment]
You may find http://astexplorer.net very useful. You can view the AST representation of your code there.

Here's the AST relevant for your question: http://astexplorer.net/#/gist/c0d0616c4ffed7af7604e5dff4cc0ce9/e325481ac69739de368c63b157b9beae7173360e

Note that `PropertyAssignment` is the `foo: bar` in `{foo: bar}`, and only in object literals.

Please note that `Lint.RuleWalker` is no longer the recommended way to traverse the AST. For performance reasons you should consider using `Lint.AbstractWalker` or `Lint.WalkContext` instead. See https://palantir.github.io/tslint/develop/custom-rules/walker-design.html
`Lint.RuleWalker` will likely be deprecated and eventually removed, maybe in TSLint v8.0

To get you started I wrote a simple walker implementation that should fit your needs:

```ts
import {isBinaryExpression, isPropertyAccessExpression, isIdentifier} from "tsutils";

function walk(ctx: Lint.WalkContext<void>) {
    return ts.forEachChild(ctx.sourceFile, function cb(node): void {
        if (isBinaryExpression(node) && node.operatorToken.kind === ts.SyntaxKind.EqualsToken && // only interested in assignments
            isPropertyAccessExpression(node.left) && node.left.name.text === "exports" && // where the assignment target is a property named `exports`
            isIdentifier(node.left.expression) && node.left.expression.text === "module") { // on an object named `module`
            // add the failure to the WalkContext
            ctx.addFailureAtNode(node, "Use 'export =' instead");
        }
        return ts.forEachChild(node, cb);
    });
}
```

**Comment by retorquere** [comment]
Perfect -- this is going to save me so much trouble. I picked up the `Lint.RuleWalker` from [here](https://palantir.github.io/tslint/develop/custom-rules/) BTW -- it's the first hit you get when you google "custom tslint rules"

**Comment by ajafff** [comment]
I know that's a bit unfortunate. The whole development documentation needs an overhaul.

I also suggest looking at the existing rules' code. You may find some useful helper functions. Especially the functions I develop in `tsutils` can save you a ton of work. But I don't have the time (or motivation) to document all of them. So the best way to learn about them is looking at their uses in tslint rules.
